experiment/manual-trigger/manual-trigger.go

```diff
@@ -0,0 +1,190 @@
+	"k8s.io/test-infra/prow/jenkins"
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/pjutil"
+)
+
+type options struct {
+	githubapiurl string
+	num          int
+	org          string
+	repo         string
+	ghtoken      string
+	jbtoken      string
+	jenkinsurl   string
+	jobname      string
+}
+
+func flagOptions() options {
+	o := options{}
+	flag.StringVar(&o.githubapiurl, "githubapiurl", "https://api.github.com", "GitHub's API endpoint.")
+	flag.StringVar(&o.jenkinsurl, "jenkinsurl", "", "Jenkins url.")
+
+	flag.StringVar(&o.ghtoken, "ghtoken", "", "Path to file containing GitHub token.")
+	flag.StringVar(&o.jbtoken, "jbtoken", "", "Path to file containing Jenkins Bearer token.")
+
+	flag.StringVar(&o.jobname, "jobname", "", "Name of Jenkins job")
+
+	flag.IntVar(&o.num, "num", 0, "GitHub issue number")
+	flag.StringVar(&o.org, "org", "", "GitHub organization")
+	flag.StringVar(&o.repo, "repo", "", "GitHub repository")
+	flag.Parse()
+	return o
+}
+
+func sanityCheckFlags(o options) error {
+	if o.num <= 0 {
+		return fmt.Errorf("empty or invalid --num")
+	}
+	if o.repo == "" {
+		return fmt.Errorf("empty --org")
+	}
+	if o.repo == "" {
+		return fmt.Errorf("empty --repo")
+	}
+	if o.ghtoken == "" {
+		return fmt.Errorf("empty --ghtoken")
+	}
+	if o.jbtoken == "" {
+		return fmt.Errorf("empty --jbtoken")
+	}
+	if o.jbtoken == "" {
+		return fmt.Errorf("empty --jobname")
+	}
+
+	_, err := url.Parse(o.githubapiurl)
+	if o.githubapiurl == "" || err != nil {
+		return fmt.Errorf("Must specify a valid URL to --githubapiurl")
+	}
+
+	_, err = url.Parse(o.jenkinsurl)
+	if o.jenkinsurl == "" || err != nil {
+		return fmt.Errorf("Must specify a valid URL to --jenkinsurl")
+	}
+
+	return nil
+}
+
+func readToken(path string) (string, error) {
+	contents, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(string(contents)), nil
+}
+
+func main() {
+	o := flagOptions()
+	err := sanityCheckFlags(o)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	ghtoken, err := readToken(o.ghtoken)
+	if err != nil {
+		log.Fatal("cannot read file specified by --ghtoken: %v", err)
+	}
+
+	jbtoken, err := readToken(o.jbtoken)
+	if err != nil {
+		log.Fatal("cannot read file specified by --jbtoken: %v", err)
+	}
+
+	c := github.NewClient(ghtoken, o.githubapiurl)
+
+	fmt.Print("Querying GitHub for details... ")
+	pr, err := c.GetPullRequest(o.org, o.repo, o.num)
+	if err != nil {
+		fmt.Print("failed!\n")
+		log.Fatal("Unable to get specified pull request: ", err)
+	}
+
+	ref, err := c.GetRef(o.org, o.repo, "heads/"+pr.Base.Ref)
+	if err != nil {
+		fmt.Print("failed!\n")
+		log.Fatal("Unable to get reference to pull request", err)
+	}
+	fmt.Print("done\n")
+
+	kr := kube.Refs{
+		Org:     o.org,
+		Repo:    o.repo,
+		BaseRef: pr.Base.Ref,
+		BaseSHA: ref,
+		Pulls: []kube.Pull{
+			{
+				Number: pr.Number,
+				Author: pr.User.Login,
+				SHA:    pr.Head.SHA,
+			},
+		},
+	}
+
+	job := config.Presubmit{
```

Comment:
Construct the `ProwJobSpec` literal here, we're not gaining much from the `PresubmitSpec()` call